example-gto/code/.github/workflows/gto-act-on-tags.yml

```diff
@@ -12,48 +12,24 @@ jobs:
     - uses: actions/checkout@v3
       with:
         fetch-depth: 0
-    - name: Fetch the whole repo in CI (including all tags and branches)
-      run: |
-        set +e
-        # https://stackoverflow.com/questions/10312521/how-to-fetch-all-git-branches
-        git tag | xargs git tag -d
-        for remote in `git branch -r`; do git branch --track ${remote#origin/} $remote; done
-        git fetch --all --prune --tags
-        git pull --all
-        set -e
+    - name: "GTO: figure out what was registered/promoted and show the Registry state"
+      id: gto
+      uses: iterative/gto-action@v1
+      with:
+        show: true
+        history: true
     - uses: actions/setup-python@v2
     - name: Install dependencies
       run: |
         pip install --upgrade pip setuptools wheel
         pip install -r requirements.txt
-    - name: "GTO: show current Registry state"
+    - name: "Publish (act on registering a new version)"
+      if: steps.gto.outputs.event == 'registration'
       run: |
-        gto show
-        gto history
-    - name: "GTO: figure out what was registered/promoted"
+        echo "[$GITHUB_REF] You got version '${{ steps.gto.outputs.version }}' registered for model '${{ steps.gto.outputs.name }}' "
+        echo "[$GITHUB_REF] It is about time to publish it somewhere so others could use it!"
+    - name: "Deploy (act on assigning a new stage)"
+      if: steps.gto.outputs.event == 'assignment'
```

Comment:
I think we should also showcase how to stop wf execution on irrelevant events. RN if you add another step after this it will be executed. And if your wf is purely "deployment", you'll have to specify `if` for each step

Reply:
Good idea. I updated the workflow, please check it out: https://github.com/aguschin/example-gto/blob/mlem/.github/workflows/gto-act-on-tags.yml

Couldn't keep it as lightweight as I would like since I had to use 2 jobs, and to pass outputs from one Job to the other, you need to define them manually. It's a limitation of GHA.